src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java

```diff
@@ -84,10 +85,48 @@ public StreamingClientProperties(Map<String, String> connectorConfig) {
         Optional.ofNullable(connectorConfig.get(SNOWPIPE_STREAMING_MAX_CLIENT_LAG));
     snowpipeStreamingMaxClientLag.ifPresent(
         overriddenValue -> {
-          LOGGER.info(
-              "Config is overridden for {}={}", SNOWPIPE_STREAMING_MAX_CLIENT_LAG, overriddenValue);
           parameterOverrides.put(MAX_CLIENT_LAG, String.format("%s second", overriddenValue));
         });
+    combineStreamingClientOverriddenProperties(connectorConfig);
+  }
+
+  private void combineStreamingClientOverriddenProperties(Map<String, String> connectorConfig) {
+    // MAX_CLIENT_LAG property can also be present in override map, in that case, we will honor the
+    // value
+    // provided in SNOWPIPE_STREAMING_MAX_CLIENT_LAG since it preceded and was released in earlier
+    // versions.
+    Optional<String> clientOverrideProperties =
+        Optional.ofNullable(connectorConfig.get(SNOWPIPE_STREAMING_CLIENT_PARAMETER_OVERRIDE_MAP));
+    Map<String, String> clientOverridePropertiesMap = new HashMap<>();
+    clientOverrideProperties.ifPresent(
+        overriddenKeyValuePairs -> {
+          LOGGER.info(
+              "Overridden map provided for streaming client Properties: {}",
+              overriddenKeyValuePairs);
+          Map<String, String> overriddenPairs =
+              Utils.parseCommaSeparatedKeyValuePairs(overriddenKeyValuePairs);
+          overriddenPairs.forEach(
+              (overriddenKey, overriddenValue) -> {
+                if (overriddenKey.equalsIgnoreCase(MAX_CLIENT_LAG)) {
+                  clientOverridePropertiesMap.put(
+                      MAX_CLIENT_LAG, String.format("%s second", overriddenValue));
```

Comment:
What is the significance of `second` here?

Reply:
Got it - I was wondering if there was a way then to special case in one place it rather than doing it when iterating over keys and then later after iterating all keys.

Reply:
umm, so this is a special case because:
1. `MAX_CLIENT_LAG` is a constant fetched from ingest sdk which is lowercase itself. In other words, this is a derived property from input snowflake.streaming.max.client.lag
2. New config added here is all customer controlled and we will always have to parse lowercase. 

Let me know if you want to discuss offline, I can add this in javadocs.

Reply:
See my comment above but TL;DR: let's add this to the Java Doc

Reply:
done!